src/program_generator/pg_validation.py

```diff
@@ -172,13 +180,22 @@ def _generate_random_arithmetic(self, opsLimit, bytecodeLimit, dominant, pushMax
       elif op in shift_ops:  # SHL, SHR, SAR need 0-255 value on the top of the stack
         bytecode += self._random_push(pushMax, randomizePush) if cleanStack or previous_nreturns == 0 else ""
         bytecode += self._random_push(1, False)
+      elif op in memory_ops:
+        # otherwise memory OPCODEs might malfunction on arbitrary args. We have only limited memory to work with.
+        assert cleanStack
```

Comment:
This is something I do not understand

Reply:
ah, I needed to think for a while, the comment needs a tweak.

It refers to the `assert`. We cannot generate programs containing memory ops unless `cleanStack`. If `not cleanStack`, a large value might appear on the stack and do something silly.